core/src/main/scala/org/apache/spark/status/storeTypes.scala

```diff
@@ -177,6 +177,10 @@ private[spark] class TaskDataWrapper(
     val accumulatorUpdates: Seq[AccumulableInfo],
     val errorMessage: Option[String],
 
+    // Non successful metrics now will have negative values in `TaskDataWrapper`.
```

Comment:
This comment seems a little bit out of place. It sounds better when merged with the comment just below the `hasMetrics` field, which also needs to be updated anyway.

Reply:
Thanks. Updated